ops/services/virtual_network/main.tf

```diff
@@ -56,13 +56,22 @@ resource "azurerm_subnet" "webapp" {
   }
 }
 
+# The name of the private DNS zone MUST be environment-specific to support multiple envs within the same resource group.
 resource "azurerm_private_dns_zone" "default" {
-  name                = "privatelink.postgres.database.azure.com"
+  name                = "privatelink.${length(trimprefix(var.env, var.env_level)) > 0 ? "${var.env}." : ""}postgres.database.azure.com"
```

Comment:
Is this true? I was under the impression that DNS zones could be shared as long as the VNet was different.

Reply:
The catch here is that, the way this was written, if we leave the name alone, TF will complain that a resource with the same name already exists in the resource group. Either we have to leverage a `data` object, and make only one of the environments in the group own the DNS zone, or we have each environment create their own. This is the solution I came up with to solve the problem of multiple environments within a single resource group.

Reply:
You could theoretically do something like this: https://stackoverflow.com/a/41870148

But otherwise I think this works, if that's the constraint we're under.

Reply:
Yeah, that's a lot more elegant...this is why I'm glad we have reviews, because my 3-AM brain never thought of that. Fixed!